src/Field.js

```diff
@@ -30,6 +31,11 @@ const createField = ({ deepEqual, getIn, setIn }) => {
         this.ConnectedField =
           createConnectedField(this.context._reduxForm, { deepEqual, getIn }, nextProps.name)
       }
+
+      const syncError = this.getSyncError()
+      if (syncError !== this.state.syncError) {
+        this.setState({ syncError })
+      }
```

Comment:
https://github.com/erikras/redux-form/pull/1120

this PR works because `Field` component always rerender.